In [1]:
import pandas as pd

In [5]:
vine_df = pd.read_csv("./data/vine_table.csv") 

vine_df

review_id  star_rating  helpful_votes  total_votes vine  \
0        R35T75OLUGHL5C          4.0            0.0          0.0    N   
1        R2BV735O46BN33          5.0            0.0          0.0    N   
2        R2NBEUGPQQGXP1          4.0            0.0          0.0    N   
3        R17LLAOJ8ITK0S          3.0            1.0          1.0    N   
4        R39PEQBT5ISEF4          1.0            0.0          0.0    N   
...                 ...          ...            ...          ...  ...   
2302396  R2LQD2IV5ZWF7G          5.0           34.0         35.0    N   
2302397  R2XLPTZ391M3PM          5.0          165.0        167.0    N   
2302398  R1BRPTHYOJF9NC          5.0           30.0         32.0    N   
2302399  R3LJVE6YICJOHI          5.0           28.0         31.0    N   
2302400   RPJDYNNK1AMKT          5.0            0.0          0.0    N   

        verified_purchase  
0                       Y  
1                       Y  
2                       Y  
3                       Y  
4                       Y  
...                   ...  
2302396                 N  
2302397                 N  
2302398                 N  
2302399                 N  
2302400                 N  

[2302401 rows x 6 columns]

In [6]:
vine_df.dtypes

review_id             object
star_rating          float64
helpful_votes        float64
total_votes          float64
vine                  object
verified_purchase     object
dtype: object

In [8]:
vine_df.info

<bound method DataFrame.info of               review_id  star_rating  helpful_votes  total_votes vine  \
0        R35T75OLUGHL5C          4.0            0.0          0.0    N   
1        R2BV735O46BN33          5.0            0.0          0.0    N   
2        R2NBEUGPQQGXP1          4.0            0.0          0.0    N   
3        R17LLAOJ8ITK0S          3.0            1.0          1.0    N   
4        R39PEQBT5ISEF4          1.0            0.0          0.0    N   
...                 ...          ...            ...          ...  ...   
2302396  R2LQD2IV5ZWF7G          5.0           34.0         35.0    N   
2302397  R2XLPTZ391M3PM          5.0          165.0        167.0    N   
2302398  R1BRPTHYOJF9NC          5.0           30.0         32.0    N   
2302399  R3LJVE6YICJOHI          5.0           28.0         31.0    N   
2302400   RPJDYNNK1AMKT          5.0            0.0          0.0    N   

        verified_purchase  
0                       Y  
1                       Y  
2      

## Step 1 
Filter the data and create a new DataFrame or table to retrieve all the rows where the total_votes count is equal to or greater than 20 to pick reviews that are more likely to be helpful and to avoid having division by zero errors later on.

In [10]:
vine_df_20 = vine_df.loc[vine_df["total_votes"] >= 20] 
vine_df_20

review_id  star_rating  helpful_votes  total_votes vine  \
65       R2FP3U4NHNFNL2          5.0           25.0         29.0    N   
159      R1UUK1977O38MU          5.0           31.0         31.0    N   
603       RXO216LWUDV6O          3.0           29.0         31.0    N   
655      R3NMJF7EBMM19V          3.0           26.0         27.0    N   
935      R2ZY0ZBDUO0XUY          3.0           20.0         21.0    N   
...                 ...          ...            ...          ...  ...   
2302395  R3MS5E3IRFK9MN          1.0            6.0         45.0    N   
2302396  R2LQD2IV5ZWF7G          5.0           34.0         35.0    N   
2302397  R2XLPTZ391M3PM          5.0          165.0        167.0    N   
2302398  R1BRPTHYOJF9NC          5.0           30.0         32.0    N   
2302399  R3LJVE6YICJOHI          5.0           28.0         31.0    N   

        verified_purchase  
65                      Y  
159                     Y  
603                     Y  
655                     Y  
935                     Y  
...                   ...  
2302395                 N  
2302396                 N  
2302397                 N  
2302398                 N  
2302399                 N  

[43574 rows x 6 columns]

## Step 2
Filter the new DataFrame or table created in Step 1 and create a new DataFrame or table to retrieve all the rows where the number of helpful_votes divided by total_votes is equal to or greater than 50%.

In [11]:
vine_df_50 = vine_df_20.loc[(vine_df_20["helpful_votes"] / vine_df_20["total_votes"]) >= .50]
vine_df_50

review_id  star_rating  helpful_votes  total_votes vine  \
65       R2FP3U4NHNFNL2          5.0           25.0         29.0    N   
159      R1UUK1977O38MU          5.0           31.0         31.0    N   
603       RXO216LWUDV6O          3.0           29.0         31.0    N   
655      R3NMJF7EBMM19V          3.0           26.0         27.0    N   
935      R2ZY0ZBDUO0XUY          3.0           20.0         21.0    N   
...                 ...          ...            ...          ...  ...   
2302394  R1V5B0I8WCU88G          5.0          327.0        339.0    N   
2302396  R2LQD2IV5ZWF7G          5.0           34.0         35.0    N   
2302397  R2XLPTZ391M3PM          5.0          165.0        167.0    N   
2302398  R1BRPTHYOJF9NC          5.0           30.0         32.0    N   
2302399  R3LJVE6YICJOHI          5.0           28.0         31.0    N   

        verified_purchase  
65                      Y  
159                     Y  
603                     Y  
655                     Y  
935                     Y  
...                   ...  
2302394                 N  
2302396                 N  
2302397                 N  
2302398                 N  
2302399                 N  

[39976 rows x 6 columns]

## Step 3
Filter the DataFrame or table created in Step 2, and create a new DataFrame or table that retrieves all the rows where a review was written as part of the Vine program (paid), vine == 'Y'.

In [12]:
paid_reviews_df = vine_df_50.loc[vine_df_50["vine"] == "Y"]
paid_reviews_df

review_id  star_rating  helpful_votes  total_votes vine  \
30395    R3KPC0NBUDASX3          5.0           25.0         25.0    Y   
206171   R119P2A95GGXX4          5.0           26.0         28.0    Y   
222172   R1HKIRME8AJ89Z          5.0           79.0         82.0    Y   
281437   R3FY3GMBGOBR22          5.0           12.0         20.0    Y   
287862   R15KH3FBSVYGBU          5.0           32.0         37.0    Y   
...                 ...          ...            ...          ...  ...   
2224140  R2S53VJNNBFALC          4.0           38.0         40.0    Y   
2225611  R1WK162P6URAST          3.0           16.0         28.0    Y   
2228972  R2YE59X5KUPARP          5.0           44.0         45.0    Y   
2232328   RU38RYOL77OVU          4.0           46.0         49.0    Y   
2232911  R1V6AIVQ232LYH          4.0           57.0         59.0    Y   

        verified_purchase  
30395                   N  
206171                  N  
222172                  N  
281437                  N  
287862                  N  
...                   ...  
2224140                 N  
2225611                 N  
2228972                 N  
2232328                 N  
2232911                 N  

[107 rows x 6 columns]

## Step 4
Repeat Step 3, but this time retrieve all the rows where the review was not part of the Vine program (unpaid), vine == 'N'.

In [14]:
unpaid_reviews_df = vine_df_50.loc[vine_df_50["vine"] == "N"]
unpaid_reviews_df

review_id  star_rating  helpful_votes  total_votes vine  \
65       R2FP3U4NHNFNL2          5.0           25.0         29.0    N   
159      R1UUK1977O38MU          5.0           31.0         31.0    N   
603       RXO216LWUDV6O          3.0           29.0         31.0    N   
655      R3NMJF7EBMM19V          3.0           26.0         27.0    N   
935      R2ZY0ZBDUO0XUY          3.0           20.0         21.0    N   
...                 ...          ...            ...          ...  ...   
2302394  R1V5B0I8WCU88G          5.0          327.0        339.0    N   
2302396  R2LQD2IV5ZWF7G          5.0           34.0         35.0    N   
2302397  R2XLPTZ391M3PM          5.0          165.0        167.0    N   
2302398  R1BRPTHYOJF9NC          5.0           30.0         32.0    N   
2302399  R3LJVE6YICJOHI          5.0           28.0         31.0    N   

        verified_purchase  
65                      Y  
159                     Y  
603                     Y  
655                     Y  
935                     Y  
...                   ...  
2302394                 N  
2302396                 N  
2302397                 N  
2302398                 N  
2302399                 N  

[39869 rows x 6 columns]

## Step 5
Determine the total number of reviews, the number of 5-star reviews, and the percentage of 5-star reviews for the two types of review (paid vs unpaid).

In [35]:
paid_5star_reviews = paid_reviews_df.loc[paid_reviews_df["star_rating"] == 5]["star_rating"].count()

paid_5star_reviews

56

In [37]:
paid_reviews = paid_reviews_df["star_rating"].count()
paid_reviews

107

In [25]:
percentage_paid_5star_reviews = round((paid_5star_reviews / paid_reviews) *100, 2)
print(f'%{percentage_paid_5star_reviews}')

%52.34


## Step 5b. Unpaid reviews

In [32]:
unpaid5star_reviews = unpaid_reviews_df.loc[unpaid_reviews_df["star_rating"] == 5]["star_rating"].count()
unpaid5star_reviews

21005

In [33]:
unpaid_reviews = unpaid_reviews_df["star_rating"].count()
unpaid_reviews

39869

In [34]:
percentage_unpaid5star_reviews = round((paid_unpaid5star_reviews / unpaid_reviews) *100, 2)
percentage_unpaid5star_reviews

52.69

## Analysis
Ho: 5star_vine = 5star_not_vine (No difference in the percentage (proportion) of 5 star ratings paid (vine) vs unpaid (not vine) 

Ha: 5star_vine <> fstar_not_vine (there is a difference)

- Percentage_5star_vine = 52.34
- percentage_5star_not_vine = 52.69
- **Finding: The percentage of 5star ratings are very similar when companing vine reviews to not_vine_reviews**
- **There is reason to believe that the percentage of 5star ratings accounts for around 50% of all reviews** 
